src/server.ts

```diff
@@ -2,16 +2,49 @@ import { ApolloServer } from "@apollo/server";
 import { startStandaloneServer } from "@apollo/server/standalone";
 import fs from "fs";
 import { resolvers } from "./resolvers.js";
+import AnalyticsDataSource from "./datasources/analytics.js";
+import { config } from "dotenv";
+
+if (process.env.DB_CONN_STRING == null) config({ path: ".env.local" });
 
 const typeDefs = fs.readFileSync("src/schema.graphql", "utf8");
 
-const server = new ApolloServer({
-  typeDefs,
-  resolvers,
-});
+export interface ContextValue {
+  dataSources: {
+    analytics: AnalyticsDataSource;
+  };
+}
+
+export const startServer = async () => {
+  const analyticsDS = new AnalyticsDataSource(process.env.DB_CONN_STRING);
+  await analyticsDS.initialize();
+
+  const server = new ApolloServer({
+    typeDefs,
+    resolvers,
+  });
+
+  const { url } = await startStandaloneServer(server, {
+    context: async () => {
+      return {
+        dataSources: {
+          analytics: analyticsDS,
+        },
+      };
+    },
+    listen: { port: 4000 },
+  });
+
+  const exitSignals = ["SIGTERM", "SIGINT"];
+  exitSignals.forEach((signal) => {
+    process.on(signal, async () => {
+      console.log(`Received ${signal}`);
+      await analyticsDS.close();
+      process.exit(0);
```

Comment:
Before the server shuts down, the connection to MongoDB (through `analytics data source`) should be closed and output [the log](https://github.com/chohanbin/flywheel-data-service/pull/4/files#r1630251970).